### PR TITLE
fix(http): throw if url is not string or Request

### LINKED ISSUE #4245: fix(http): should throw an exception if input to `http.request` is not string or `Request`
I forgot to import `Request` in an example where I instantiate `Request` manually, which meant I was instantiating the global `Request` object instead of `Request` provided by `angular2/http`. When passing this to `Http.prototype.request`, the input didn't match the string or `instanceof Request` check, so the method was returning `undefined. Instead, an exception should be raised.

### PR DESCRIPTION
**Pull request by bkyarger**
Closes #4245

### CODE REVIEW

**Comment by jeffbcross** [comment]
@bkyarger FYI we're having issues with our CI environment right now, so some failures are expected. I'll restart jobs after the issues are resolved.

**Comment by bkyarger** [comment]
@jeffbcross sounds great.  I'll check back tonight to see if there are still CI issues.

File: modules/angular2/test/http/http_spec.ts [hunk 1]
```diff
@@ -160,6 +160,12 @@ export function main() {
         //         async.done();
         //       });
         //     }));
+
+
+        it('should throw if url is not a string or Request', () => {
+          var req = <Request>{};
+          expect(() => http.request(req)).toThrowError();
```
Comment (jeffbcross) [inline modules/angular2/test/http/http_spec.ts]: Could you add the expected message here? `toThrowError('First argument must be a string or Request.')`

File: modules/angular2/src/http/http.ts [hunk 2]
```diff
@@ -120,6 +120,8 @@ export class Http {
           new Request(mergeOptions(this._defaultOptions, options, RequestMethods.Get, url)));
     } else if (url instanceof Request) {
       responseObservable = httpRequest(this._backend, url);
+    } else {
+      throw makeTypeError("First argument must be a string or Request.");
```
Comment (jeffbcross) [inline modules/angular2/src/http/http.ts]: A couple of tweaks: 
 * Use single quotes for consistency
 * Could you add the same logic to `Jsonp` in the same file?
 * Could you say "must be a __url__ string or Request __instance__"?

Comment (bkyarger) [inline modules/angular2/src/http/http.ts]: Should I add a test for ```Jsonp```?  I don't see any existing tests for it.

File: modules/angular2/test/http/http_spec.ts (hunk 1 above)
Comment (bkyarger) [inline modules/angular2/test/http/http_spec.ts]: Done

File: modules/angular2/src/http/http.ts (hunk 2 above)
Comment (bkyarger) [inline modules/angular2/src/http/http.ts]: Done, if you want tests added for ```Jsonp``` as well let me know.

File: modules/angular2/test/http/http_spec.ts [hunk 3]
```diff
@@ -160,6 +160,13 @@ export function main() {
         //         async.done();
         //       });
         //     }));
+
+
+        it('should throw if url is not a string or Request', () => {
+          var req = <Request>{};
+          expect(() => http.request(req))
```
Comment (jeffbcross) [inline modules/angular2/test/http/http_spec.ts]: I'm surprised we don't have a Jsonp unit test for this method! Could you just duplicate this expectation in the same spec and call `jsonp.request`?

Comment (bkyarger) [inline modules/angular2/test/http/http_spec.ts]: Yep, will do.

Comment (bkyarger) [inline modules/angular2/test/http/http_spec.ts]: Added a new describe block for Jsonp, along with this same expectation for jsonp.

**Comment by jeffbcross** [comment]
Awesome! LGTM. I'll try to merge it...our CI woes only affect PRs at the moment, so it should merge without issue.

**Pull request closed without merging**